### DISCUSSION TITLE
ESP32 lightsleep() WDT conundrum

### DISCUSSION THREAD

**Discussion by davefes** [micropython/micropython#10173] [Using MicroPython] [answered]
I want to put an ESP32 into lightsleep() and also to check that it is still
running say, once a minute.

I want the system WDT and/or a SW WDT to bring the unit out of lightsleep(),
if something "hangs", BUT I do not want to run the code if it is woken-up just
to check if it is still alive.

In:
https://docs.micropython.org/en/latest/library/machine.html#machine-constants
and for deepsleep() there is a mention of:

> With or without a timeout, execution may resume at any time if there are events
> that require processing. Such events, or wake sources, should be configured before
> sleeping, like Pin change or RTC timeout.

I already use Pin change to trigger running the code to take a picture.  

I understand that RTC timeout just refers to the value you put in lightsleep(), correct?

I guess what I would like is when lightsleep(60000) times-out that I get a different
machine.reset_cause() then when it is woken-up by a:
`   esp32.wake_on_ext0()`

Any suggestions welcomed.

Is there anything I could do with RTC.alarm or RTC.irq()?

**Comment by davefes** [comment, marked as answer]
Wasn't so difficult after all:
```py
import esp32
from machine import Pin, lightsleep
import utime


wake_source = Pin(27, Pin.IN) #  setting wake up pin, has a 10K to ground
esp32.wake_on_ext0(pin = wake_source, level = esp32.WAKEUP_ANY_HIGH)


print ('waiting for 5 seconds to allow a CTRL-C')
utime.sleep(5)

while True:

    if (wake_source.value() == 1):
        print ('we got a valid trigger')
        utime.sleep(0.1)

     #  do the processing you want

 #  feed the watchdog(s)

    print ('going to lightsleep for 60 seconds')
    utime.sleep(0.1)

    lightsleep(60000)
```
BTW, it looks RTC.irq() is not in v1.19.1

**Reply by scruss** [reply]
`machine.RTC.irq()` isn't available in all ports. The documentation's a bit light on that
